### PR TITLE
Update resource declaration to support AWS provider 4.x

### PR DESCRIPTION
**Pull request by chrischildresssg**
## Change description

> Replace deprecated `aws_s3_bucket_object` resource declaration with the newer `aws_s3_object` resource declaration.

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#3 ]() 

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## List of backwards incompatible changes
+
+- `aws_s3_object` was added in AWS provider v4 and is not supported in AWS provider versions 3.74.2 or earlier
+
+### State Changes
+
+To migrate from the `v1.x` version to `v2.x` version of the module, the following state remove and import commands can be performed to maintain the current resources without unnecessary modification:
+
+```bash
+terraform import 'module.ecs_service.aws_s3_object.artifacts_s3' my-artifact-bucket/artifacts.zip
```
Comment (wesleykirklandsg) [inline UPGRADE-2.0.md:18]: Nice comment, I wouldn't have thought about that myself.

**Review by wesleykirklandsg — CHANGES_REQUESTED** [review]
@chrischildresssg There isn't a provider definition file. One needs to exist for this to be a valid change. See here for the base standards.

https://github.com/StratusGrid/template-terraform-module-starter

File: UPGRADE-2.0.md (hunk 1 above)
Comment (chrischildresssg) [reply UPGRADE-2.0.md:18]: That is now complete. I've also renamed the hyphenated files.

**Comment by chrischildresssg** [comment]
> @chrischildresssg There isn't a provider definition file. One needs to exist for this to be a valid change. See here for the base standards.
> 
> https://github.com/StratusGrid/template-terraform-module-starter

This is now complete.

**Review by wesleykirklandsg — APPROVED** [review]
LGTM

**Pull request merged**